pkg/common/cliprinter/internal/protojson/protojson.go

```diff
@@ -34,13 +34,66 @@ func Print(msgs []proto.Message, stdout, stderr io.Writer) error {
 
 		jms = append(jms, jb)
 	}
-
 	var err error
-	if len(jms) == 1 {
-		err = json.NewEncoder(stdout).Encode(jms[0])
+
+	parsedJms, err := parseJSONMessages(jms)
+	if err != nil {
+		_ = errorjson.Print(err, stdout, stderr)
+	}
+
+	if len(parsedJms) == 1 {
+		err = json.NewEncoder(stdout).Encode(parsedJms[0])
 	} else {
-		err = json.NewEncoder(stdout).Encode(jms)
+		err = json.NewEncoder(stdout).Encode(parsedJms)
 	}
 
 	return err
 }
+
+func parseJSONMessages(jms []json.RawMessage) ([]json.RawMessage, error) {
+	var parsedJms []json.RawMessage
+	for _, jm := range jms {
+		parsedJm, err := parseJSONMessage(jm)
+		if err != nil {
+			return nil, err
+		}
+		parsedJms = append(parsedJms, parsedJm)
+	}
+
+	return parsedJms, nil
+}
+
+func parseJSONMessage(jm json.RawMessage) (json.RawMessage, error) {
+	var jmMap map[string]interface{}
+	if err := json.Unmarshal(jm, &jmMap); err != nil {
+		return nil, err
+	}
+
+	removeNulls(jmMap)
+
+	parsedJm, err := json.Marshal(jmMap)
+	if err != nil {
+		return nil, err
+	}
+
+	return parsedJm, nil
+}
+
+func removeNulls(jsonMap map[string]interface{}) {
+	for key, val := range jsonMap {
+		if val == nil {
+			delete(jsonMap, key)
+			continue
+		}
+		switch v := val.(type) {
```

Comment:
Can this be any type other than the two listed?

Reply:
According to the `json.Unmarshal` documentation, when unmarshalling data into a interface{} value, Unmarshal stores one of these in the interface value:
```go
//
//	bool, for JSON booleans
//	float64, for JSON numbers
//	string, for JSON strings
//	[]interface{}, for JSON arrays
//	map[string]interface{}, for JSON objects
//	nil for JSON null
```
Since this method only looks for removing null values, we only have to check for nil values, or values that can contain nil values, like a `map[string]interface{}` or `[]interface{}`, so this switch clause, with the nil check above it, should be sufficient for this use case.

Reply:
Got it, the other values are ok to leave as is.